yew/src/services/websocket.rs

```diff
@@ -172,13 +180,33 @@ impl WebSocketService {
         &mut self,
         url: &str,
         notification: &Callback<WebSocketStatus>,
-    ) -> Result<ConnectCommon, &str> {
+    ) -> Result<ConnectCommon, WebSocketError> {
         let ws = WebSocket::new(url);
-        if ws.is_err() {
-            return Err("Failed to created websocket with given URL");
+        #[cfg(feature = "web_sys")]
+        {
+            if let Err(ws_error) = &ws {
+                return Err(WebSocketError::CreationError(
+                    ws_error
+                        .clone()
+                        .unchecked_into::<js_sys::Error>()
+                        .to_string()
+                        .as_string()
+                        .unwrap(),
+                ));
+            };
+        }
+        #[cfg(feature = "std_web")]
+        {
+            if ws.is_err() {
+                return Err(WebSocketError::CreationError(
+                    "Error opening a WebSocket connection.".to_string(),
+                ));
+            };
         }
 
-        let ws = ws.map_err(|_| "failed to build websocket")?;
+        let ws =
+            ws.map_err(|_| WebSocketError::CreationError("Failed to build websocket".into()))?;
```

Comment:
I just realized that the previous code was checking the same result twice. We should consolidate :)

Can you move the `web-sys` and `stdweb` checks inside this `map_err`? Actually, a `match` statement would probably be better. The default branch could be an assert false statement

Reply:
Yes. I was a bit confused by this too. I'll match against the error inside the `map_err` closure.